hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/crypto/key/kms/ValueQueue.java

```diff
@@ -299,19 +299,18 @@ public E getNext(String keyName)
    * @param keyName the key to drain the Queue for
    */
   public void drain(String keyName) {
+    Runnable e;
+    while ((e = queue.deleteByName(keyName)) != null) {
+      executor.remove(e);
+    }
+    writeLock(keyName);
     try {
-      Runnable e;
-      while ((e = queue.deleteByName(keyName)) != null) {
-        executor.remove(e);
-      }
-      writeLock(keyName);
-      try {
-        keyQueues.get(keyName).clear();
-      } finally {
-        writeUnlock(keyName);
+      LinkedBlockingQueue kq = keyQueues.getIfPresent(keyName);
+      if (kq != null) {
+        kq.clear();
       }
-    } catch (ExecutionException ex) {
```

Comment:
Hi @xiaoyuyao , Not sure if it is compatible to throw out this exception. Other are fair enough to me. Thanks.

Reply:
We don't have compatibility issue here. LoadingCache#get() may throw ExecutionException. But this ticket has change to use LoadingCache#getIfPresent() which does not throw ExecutionException. This way, we don't need to worry about either NPE from LoadingCache.get().clear() or ExecutionException.